### ISSUE TITLE
Switchbot BOT and homey

### ISSUE THREAD

**Issue by mult1pl1c1ty** [AdyRock/com.switchbot#114]
When I connect the switchbot bot to homey (no Switchbot hub) then the switchbot app workds perfectly and the Homey hub works, but takes approx 20 seconds each time I press. Seems it's struggling to connect each time? Any ideas?

**Comment by AdyRock** [comment]
Homey's BLE is not great, so that is normal unfortunately.

**Comment by mult1pl1c1ty** [comment]
Damn. Doesn't look like I'll be using switch bit then. Cheers for the quick
response

**Comment by AdyRock** [comment]
It is much more responsive with the hub, typically 1 to 3 seconds to respond.

**Comment by mult1pl1c1ty** [comment]
Ok, thanks.

**Comment by mult1pl1c1ty** [comment]
Any chance the app will support the Hub mini anytime in the future?

**Comment by AdyRock** [comment]
It already supports it. As it has no capabilities of its own, you can't add it to Homey, but any devices connected to it will be available.

**Comment by mult1pl1c1ty** [comment]
Oh. How do I add it? I can’t see it in the list of devices?

**Comment by mult1pl1c1ty** [comment]
Sorry. I’ve just re read your email. Doh. I’ll give it a look

**Comment by mult1pl1c1ty** [comment]
I guess I just do new device and select the bot? I’ve tried this but it
seems to totally ignore the hub and just add the switch to bot as if there
isn’t any hub? what am I doing wrong?

**Comment by AdyRock** [comment]
Providing you select the Bot (hub) driver and it finds it then it's working. You don't see the hub at all but Homey sends commands to the cloud, which then sends the commands to the hub and that sends the commands to the Bot. 
I have just noticed the Bot (hub) is in the Wall Plug section.

**Comment by mult1pl1c1ty** [comment]
I just get "no new devices have been found"

I'll try again tomorrow...

**Comment by AdyRock** [comment]
Have you enabled cloud services for the Bot in the Switchbot app?
Could you send the Detected log from the app settings page.

**Comment by mult1pl1c1ty** [comment]
Yes, I did! I'll send it soon...

**Comment by mult1pl1c1ty** [comment]
Ok. So not much point me sending the log because I've currently got the
switch running through alexa. I'm pretty sure the problem all stems from
the fact that homey cant see the hub (so I can't add it as a device).
Consequently when I add the switch its always adding it as BLE only...

**Comment by AdyRock** [comment]
No, as I said, Homey doesn't need to know about the hub as it just sends commands to the cloud. SwitchBot then sends the command to the hub that controls the device.
If you send the log I can see if the Bot is available via the cloud. 
When you try to add the Bot (hub) device, do you successfully login to your SwitchBot account as part of the process?

**Comment by mult1pl1c1ty** [comment]
I think we may be at cross purposes (my fault)

Let me try to clarify
 When you say

"When you try to add the Bot (hub) device"

I presume you mean ad to the switchbot app? If so, then yes this all works
and I do indeed successfully login to my switchbot account

If, however you mean do I add it to homey (as switchbot/ BOT HUB device IS
one of the available selections in the switchbot app within homey) then NO
- Homey doesn't find it.

I've also added the Switchbot APP logs for you (from the Switchbot app/Hub
Mini/Settings/Logs

Let me know if you need anything else? and thanks for helping.

**Comment by AdyRock** [comment]
Unfortunately, I don't get the logs from the SwitchBot mobile app.

I do mean when you add the Bot to Homey.
<img width="289" alt="image" src="https://github.com/AdyRock/com.switchbot/assets/38184396/ad62eb69-44b9-4301-9814-89cd7ccac1ed">
When you select this, it should ask you to log into your SwitchBot account (unless you have already added a device via the cloud). Once you have logged in, you should then see the device.

The log I need is from the Homey - SwitchBot app - settings page - Detected tab
<img width="242" alt="image" src="https://github.com/AdyRock/com.switchbot/assets/38184396/9218f214-b68d-4a49-9da2-5ec8fdee33a7">
Click on Get HUB devices and once the list is full, tap on Send Log.

**Comment by mult1pl1c1ty** [comment]
Cool. So.


When I try to add the bot to Homey I do indeed get a sign in page (which I
do and the page says 'you can close this page now") but then Homey just
says it can't find any devices??

**Comment by mult1pl1c1ty** [comment]
So I 'think' I've got it working. Totally deleted the hub mini and switch,
then deleted my account from Switchbot and started all over again.

When everything was re-added I did as you said, added the switch BOT hub
(signed in but Homey still didn't find it) then added the switch. It's
definitely NOT BLE connected (A 'w' shows up in the homey app ICON) and it
seems to work. It is quite slow mind you but I presume that's normal?

Thanks for the help.

G

**Comment by mult1pl1c1ty** [comment]
Am I correct in thinking that even though there are flow cards for turn on
and off, the switchbot switch doesn't actually know whether it's on or off
anyway???

**Comment by AdyRock** [comment]
I think it does know if it's on or off. All my devices are currently packed in boxes while I redecorate, so I can't test it ATM. 
If you look at the on/off status in the Homey device and change the Bot via the Switchbot app, does the Homey device update?
